src/core/server/saved_objects/service/lib/search_dsl/search_dsl.ts

```diff
@@ -66,11 +66,15 @@ export function getSearchDsl(
     throw Boom.notAcceptable('sortOrder requires a sortField');
   }
 
+  if (namespaces?.some((namespace) => namespace.indexOf('*') >= 0)) {
+    throw Boom.notAcceptable(`namespaces cannot contain wildcards ("*")`);
+  }
```

Comment:
> One thing that sticks out to me here is that we're effectively changing the valid API parameters based on whether or not the Spaces plugin is enabled. We rely on the SpacesSavedObjectsClient wrapper to transform a "*" into an array of space IDs

Yeah, that's a good point.

> Perhaps instead we should treat "*" as the Default space here? Then we would have the following cases when searching with "*"

I'm torn on this. Conceptually it makes sense, as the Default space is the only real namespace that exists when the Spaces plugin is disabled. We could alternatively pass this through unmodified, which would just return 0 objects, since there won't ever be a namespace which contains a wildcard. I double checked the terms query, and it does not appear to interpret wildcards, so passing this through shouldn't return unexpected objects. My intent for putting this in was prevent the wildcard from being interpreted, but I think that was premature.

So, I think we have three options:
1) Leave this alone, and continue to disallow `*` 
2) Change `*` to `undefined`/`default` to represent the Default space
3) Allow the namespaces through as-is. Don't block or rename the namespaces.

I _think_ I'm favoring option 3, but I'm interested in hearing your thoughts too.

I like option 2 because the APIs will be compatible between the OSS and Default distributions, but there's a part of me that worries about silently interpreting `*` as the default space here. A bug in the already complex saved objects service could inadvertently cause us to return objects from the Default Space when the user isn't actually authorized to access them.

Reply:
> We could alternatively pass this through unmodified, which would just return 0 objects, since there won't ever be a namespace which contains a wildcard.

This is true currently, but we might eventually support a `"*"` in a multi-namespace object's `namespaces` array with #58756.

> there's a part of me that worries about silently interpreting * as the default space here. A bug in the already complex saved objects service could inadvertently cause us to return objects from the Default Space when the user isn't actually authorized to access them.

Yeah, that's certainly a valid concern. It would only be a problem if that bug managed to happen while Spaces and Security are enabled, right? 

All of the logic combined is pretty complex, but the Spaces SO client code for checking for a wildcard is pretty simple.

Ultimately I'd still lean towards option 3, but leaving option 1 in place is probably fine as this is a corner case.

Reply:
> This is true currently, but we might eventually support a "*" in a multi-namespace object's namespaces array with #58756.

Great point! Thanks for the reminder.

> Yeah, that's certainly a valid concern. It would only be a problem if that bug managed to happen while Spaces and Security are enabled, right?

Yes, that's right. For what it's worth, I expect that this scenario (spaces + security enabled) is one of the more common setups these days, especially since both are free.

> Ultimately I'd still lean towards option 3, but leaving option 1 in place is probably fine as this is a corner case.

Let me see what option 3 looks like and we can decide

Reply:
@jportner I think I like Option 3 as well. Take a look and let me know what you think too: [`602d8ff` (#67644)](https://github.com/elastic/kibana/pull/67644/commits/602d8ff7d611e0c46bbb29ac231af6a7606eb652)

Reply:
> Take a look and let me know what you think too:

Love it!

Reply:
I think I like Option 3, but it seems like you implemented Option 2 ;)

Reply:
> I think I like Option 3, but it seems like you implemented Option 2 😉

Ha, you're right, sorry about that. If we go with option 3, then searching using the `*` will return 0 results for the OSS distribution, and the Default distribution when Spaces is disabled. Are you comfortable with that inconsistency?

In reality, the `namespaces` option shouldn't be used in OSS, so I'm not _terribly_ concerned about that scenario. The default distribution with Spaces disabled has the potential to be problematic though. With Option 3, consumers will have to check to see if Spaces is enabled first before using the `?namespaces=*` parameter. We have a hard enough time getting teams to think about Spaces at all, let alone the scenario where Spaces may be disabled.

Reply:
I'd like for OSS Kibana Platform to be a neutral foundation for third-party plugins to build on top off, but not breaking plugins when Spaces is disabled sounds like a good idea. So happy to go with Option 2. If a third-party developer ever opens an issue we can change this behaviour ;-)